app/views/handlers/primary_person_list_collector.py

```diff
@@ -26,18 +26,20 @@ def get_context(self):
         return build_question_context(self.rendered_block, self.form)
 
     def handle_post(self):
+        list_name = self.rendered_block['for_list']
+        self.evaluate_and_update_section_status_on_list_change(list_name)
```

Comment:
This needs to done after the action such as once the person is added or removed.